File: Readability (Python)/readability.py
import cs50
letters = 0
words = 1
sentences = 0
text = cs50.get_string("Text: ")
for i in range(0, len(text)):
    if text[i] >= 'a' and text[i] <= 'z' or text[i] >= 'A' and text[i] <= 'Z':
        letters = letters + 1
    elif text[i] == ' ':
        words = words + 1
    elif text[i] == '.' or text[i] == '!' or text[i] == '?':
        sentences = sentences + 1
L = (letters * 100) / words
S = (sentences * 100) / words
index = 0.0588 * L - 0.296 * S - 15.8
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: C Applications/Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    double letters = 0;
    double words = 1;
    double sentences = 0;
    string text = get_string("Text: ");
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }
    for (int j = 0, n = strlen(text); j < n; j++)
    {
        if (text[j] == ' ')
        {
            words++;
        }
    }
    for (int k = 0, n = strlen(text); k < n; k++)
    {
        if (text[k] == '.' || text[k] == '!' || text[k] == '?')
        {
            sentences++;
        }
    }
    double L = letters * 100 / words;
    double S = sentences * 100 / words;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %0.0f\n", index);
    }
}